### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by EMH333**
Adds the initial AlmaLinux repos. Uses resources like [`yum-remi-chef`](https://github.com/sous-chefs/yum-remi-chef) instead of attributes like [`yum-centos`](https://github.com/sous-chefs/yum-centos).

### CODE REVIEW

**Comment by EMH333** [comment]
@ramereth This is ready for an initial review. Still need to add documentation and add more to both the chefspec and inspec tests, but I'd appreciate a look at this stage!

File: metadata.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name 'yum-almalinux'
+maintainer        'Sous Chefs'
+maintainer_email  'help@sous-chefs.org'
+license           'Apache-2.0'
+description 'Installs/Configures AlmaLinux yum repositories'
+version '0.1.0'
+chef_version '>= 16.0'
```
Comment (ramereth) [inline metadata.rb:7]: Please align the text fields.

File: kitchen.dokken.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  name: dokken
+  privileged: true
+
+transport:
+  name: dokken
+
+provisioner:
+  name: dokken
+  chef_version: <%= ENV['CHEF_VERSION'] || 'current' %>
+  chef_license: accept-no-persist
```
Comment (ramereth) [inline kitchen.dokken.yml:12]: These actually needs to be moved into the `driver` section. For whatever reason, this is how the dokken plugin works.

File: metadata.rb (hunk 1 above)
Comment (ramereth) [inline metadata.rb:7]: This is the first version which included support for identifying AlmaLinux in ohai.
```suggestion
chef_version '>= 16.10'
```

File: spec/unit/recipes/default_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
+require 'spec_helper'
+
+describe 'yum-almalinux::default' do
+  step_into :yum_alma_baseos
+  step_into :yum_alma_appstream
+  step_into :yum_alma_extras
+
+  %w(8).each do |v|
```
Comment (ramereth) [inline spec/unit/recipes/default_spec.rb:8]: I'll work on adding 9 to fauxhai-ng but won't be included for a while. You should add a `TODO` comment here so we can add that once it's done.

File: spec/unit/recipes/default_spec.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        it do
+          expect(chef_run).to create_yum_repository('appstream')
+            .with(mirrorlist: 'http://mirrorlist.centos.org/?release=$releasever&arch=$basearch&repo=AppStream')
+        end
+        it do
+          expect(chef_run).to create_yum_repository('extras')
+            .with(mirrorlist: 'http://mirrorlist.centos.org/?release=$releasever&arch=$basearch&repo=extras')
+        end
+      end
+    end
```
Comment (ramereth) [inline spec/unit/recipes/default_spec.rb:44]: Can you please make a new context which tries the various properties to make sure they work properly?

File: test/integration/default/controls/default_control.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  crb
+  highavailability
+  nfv
+  plus
+  powertools
+  resilientstorage
+  rt
+  sap
+  saphana
+  ).each do |name|
```
Comment (ramereth) [inline test/integration/default/controls/default_control.rb:35]:
```suggestion
  %w(
    crb
    highavailability
    nfv
    plus
    powertools
    resilientstorage
    rt
    sap
    saphana
  ).each do |name|
```

**Review by ramereth — CHANGES_REQUESTED** [review]

File: documentation/yum_alma_appstream.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+| `debug_baseurl`     | `String`        | `https://repo.almalinux.org/...`, see [`alma_repo_baseurl`](../libraries/helpers.rb)       |
+| `debug_description` | `String`        | See [`alma_repo_description`](../libraries/helpers.rb)                                     |
+| `debug_enabled`     | `true`, `false` | `false`                                                                                    |
+| `gpgkey`            | `String`        | `https://repo.almalinux.org/...`, see [`alma_gpg_key`](../libraries/helpers.rb)            |
+| `gpgcheck`          | `true`, `false` | `true`                                                                                     |
+
+The `extra_options` property allows setting additional settings on the internal [`yum_repository`](https://docs.chef.io/resources/yum_repository/) resources.
+
+| Name          | Type | Default |
+| ------------- | ---- | ------- |
```
Comment (ramereth) [inline documentation/yum_alma_appstream.md:32]: You need to fix the alignment here, please do this globally if it needs it.

**Review by ramereth — COMMENTED** [review]
I think after this it should be ready to go!

File: documentation/yum_alma_appstream.md (hunk 6 above)
Comment (EMH333) [reply documentation/yum_alma_appstream.md]: Good catch!

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [1.0.0](https://github.com/sous-chefs/yum-almalinux/releases/tag/1.0.0)

**Pull request merged**